tests/unit/test__struct.py

```diff
@@ -0,0 +1,150 @@
+# Copyright (c) 2017 The sqlalchemy-bigquery Authors
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of
+# this software and associated documentation files (the "Software"), to deal in
+# the Software without restriction, including without limitation the rights to
+# use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+# the Software, and to permit persons to whom the Software is furnished to do so,
+# subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+# FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+import datetime
+
+import pytest
+
+import sqlalchemy
+
+
+def _test_struct():
+    from sqlalchemy_bigquery import STRUCT
+
+    return STRUCT(
+        name=sqlalchemy.String,
+        children=sqlalchemy.ARRAY(
+            STRUCT(name=sqlalchemy.String, bdate=sqlalchemy.DATE)
+        ),
+    )
+
+
+def test_struct_colspec():
+    assert _test_struct().get_col_spec() == (
+        "STRUCT<name STRING, children ARRAY<STRUCT<name STRING, bdate DATE>>>"
+    )
+
+
+def test_struct_repr():
+    assert repr(_test_struct()) == (
+        "STRUCT(name=String(), children=ARRAY(STRUCT(name=String(), bdate=DATE())))"
+    )
+
+
+def test_bind_processor():
+    assert _test_struct().bind_processor(None) is dict
+
+
+def _col():
+    return sqlalchemy.Table(
+        "t", sqlalchemy.MetaData(), sqlalchemy.Column("person", _test_struct()),
+    ).c.person
+
+
+@pytest.mark.parametrize(
+    "expr,sql",
+    [
+        (_col()["name"], "`t`.`person`.name"),
+        (_col()["Name"], "`t`.`person`.Name"),
+        (_col().NAME, "`t`.`person`.NAME"),
+        (_col().children, "`t`.`person`.children"),
+        (
+            _col().children[0].label("anon_1"),  # SQLAlchemy doesn't add the label
```

Comment:
Should we file an issue for this to investigate later? If so, let's add `TODO` and link to the issue.

Reply:
Done